relentless/core/variable.py

```diff
@@ -370,10 +373,10 @@ def _derivative(self, param):
         pass
 
     @classmethod
-    def _assert_variable(cls, v):
-        #Checks if the dependent variable depends on another variable.
-        if not isinstance(v, Variable):
-            raise TypeError('Dependent variables can only depend on other variables.')
+    def _make_variable(cls, v):
+        #Casts a scalar into an independent variable.
+        if np.isscalar(v):
+            v = IndependentVariable(value=v)
```

Comment:
After you do this conversion, you should probably keep the:
```py
if not isinstance(v,Variable):
    raise ...
```
For example, if `v` is a list, it will not get converted to an `IndependentVariable`, but the code will silently proceed.